src/components/feedback-particle-effect/feedback-text-effects.ts

```diff
@@ -1,159 +1,63 @@
-import { TextParticle } from "@feedbackParticleEffect/text-particle";
-import { font } from "@common";
+import { hideShowElement, lang, Utils } from "@common";
+import { feedbackTextDefault } from "@constants";
+import { feedbackCustomFonts } from "@data/feedback-fonts";
 
 export class FeedbackTextEffects {
-  public context: CanvasRenderingContext2D;
-  public canvasWidth: number;
-  public canvasHeight: number;
-  public textX: number;
-  public textY: number;
-  public fontSize: number;
-  public lineHeight: number;
-  public maxTextWidth: number;
-  public particleDuration: number;
-  public startTime: number | null;
-  public particles: TextParticle[];
-  public gap: number;
-  public mouse: { radius: number; x: number; y: number };
-  public textWorker: Worker;
+  private feedbackTextElement: HTMLElement | null;
 
-  constructor(
-    context: CanvasRenderingContext2D,
-    canvasWidth: number,
-    canvasHeight: number
-  ) {
-    this.context = context;
-    this.canvasWidth = canvasWidth;
-    this.canvasHeight = canvasHeight;
-    this.textX = this.canvasWidth / 1.8;
-    this.textY = this.canvasHeight / 2;
-    this.fontSize = this.canvasWidth / 7.5;
-    this.lineHeight = this.fontSize * 0.8;
-    this.maxTextWidth = this.canvasWidth * 5;
-    this.particleDuration = 5000;
-    this.startTime = null;
-    this.particles = [];
-    this.gap = 3;
-    this.mouse = { radius: 2000, x: 0, y: 0 };
-    this.textWorker = new Worker(window.feedbackTextWorkerPath);
-    this.textWorker.addEventListener(
-      "message",
-      this.handleTextWorkerMessage.bind(this)
-    );
-  }
+  constructor() {
+    this.feedbackTextElement = document.getElementById("feedback-text");
 
-  public wrapText(text: string): void {
-    const gradient = this.context.createLinearGradient(
-      0,
-      0,
-      this.canvasWidth,
-      this.canvasHeight
-    );
-    gradient.addColorStop(0.3, "#F8E218");
-    gradient.addColorStop(0.5, "#F8E218");
-    gradient.addColorStop(0.7, "#E39D37");
-    this.context.fillStyle = gradient;
-    this.context.textAlign = "center";
-    this.context.textBaseline = "middle";
-    this.context.lineWidth = 3;
-    this.context.strokeStyle = "#A46225";
-    this.context.font = `${
-      this.fontSize - text.length * 0.3
-    }px ${font}, monospace`;
-    // break multiline text
-    let lineArray: string[] = [];
-    let words = text.split(" ");
-    let lineCounter = 0;
-    let line = "";
-    for (let i = 0; i < words.length; i++) {
-      let testLine = line + words[i] + " ";
-      if (this.context.measureText(testLine).width > this.maxTextWidth) {
-        line = words[i] + "";
-        lineCounter++;
-      } else {
-        line = testLine;
-      }
-      lineArray[lineCounter] = line;
+    if (!this.feedbackTextElement) {
+      console.error("Feedback text element not found!");
+      return;
```

Comment:
Is it a good practice to add this in the constructor?

Reply:
@bernhardccodev made a changes here and keeps the constructor clean and separates initialization logic into its own method. please check